typescript/packages/plugins/allora/src/allora.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Tool } from '@goat-sdk/core'
+import { createToolParameters } from '@goat-sdk/core'
+import { z } from 'zod'
+import { AlloraAPIClient } from './api'
+
+export class AlloraService {
+    private readonly client: AlloraAPIClient
+
+    constructor(apiKey: string) {
+        this.client = new AlloraAPIClient(apiKey)
+    }
+
+    @Tool({
+        description: 'Fetch a future price prediction for a crypto asset from Allora Network.  Specify 5 minutes from now `5m`, or 8 hours from now `8h`.',
+    })
+    async getPricePrediction(params: GetAlloraPricePredictionParams) {
+        const { ticker, timeframe } = params
+        const response = await this.client.fetchAlloraPricePrediction(ticker, timeframe)
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`)
+        }
+        return await response.json()
+    }
+}
+
+const alloraPricePredictionSchema = z.object({
+    ticker: z.enum(['BTC', 'ETH']).describe('The ticker of the currency for which to fetch a price prediction.'),
+    timeframe: z.enum(['5m', '8h']).describe('The timeframe for the prediction (currently, either "5m" or "8h").'),
+})
```

Comment:
perhaps this is also for further out, but we can expose a `topics` endpoint from our indexer/api and then let the AI figure out which topic or set of topics to use

Reply:
if we used the direct RPC for getting predictions we could support this today.

we could do so in a more sustainable way through the API too, just a little lift to build the consumer for it

Reply:
Noted